src/test/java/org/junitpioneer/jupiter/StdIoExtensionTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.engine.AbstractJupiterTestEngineTests;
+import org.junitpioneer.jupiter.StdIOExtension.StdIn;
+import org.junitpioneer.jupiter.StdIOExtension.StdOut;
+
+@ExtendWith(StdIOExtension.class)
+@DisplayName("StdIOExtension ")
+public class StdIoExtensionTests extends AbstractJupiterTestEngineTests {
+
+	final BasicCommandLineApp app = new BasicCommandLineApp();
+
+	@Nested
+	@DisplayName("with specific configuration ")
+	class ConfigurationTests {
+
+		@Test
+		@DisplayName("fails if parameter is not annotated with @StdIntercept")
+		void needsAnnotation() {
+			try {
+				executeTestsForMethod(StdIOExtensionConfigurationTests.class, "noAnnotation");
+			}
+			catch (ParameterResolutionException ex) {
+				assertThat(ex.getMessage()).startsWith("No ParameterResolver registered");
+			}
```

Comment:
Since JUnit5 `assertThrows` should be used, when exceptions are expected. While this method itself can hold the expected exception I personal like to use an super class here and to check for the exception type later using _AssertJ_. Taking this test as an example it would look like this:

```
Throwable thrown = assertThrows(Throwable.class, () -> 
  // Call your system under test here
  executeTestsForMethod(StdIOExtensionConfigurationTests.class, "noAnnotation");
);

String expectedMessage = "No ParameterResolver registered";

assertThat(thrown).isExactlyInstanceOf(ParameterResolutionException.class);
assertThat(thrown.getMessage()).startsWith(expectedMessage);

```